### ISSUE TITLE
groups in 'set' notation are not translated to 'structured' config

### ISSUE THREAD

**Issue by MicLennartz** [codeout/junoser#12]
I would like to use/support this project and came across a significant problem.
Configurations in 'set' notations, using groups, are not properly translated to structured config:

```
echo "set groups VPN01894 snmp location 'room1'" | junoser -s
groups {
    VPN01894 snmp location 'room1';
}
```

Any help or hint around that would be very appreciated.
//Michael

**Comment by codeout** [comment]
Thanks for your report. It's known behavior.

We need special treatment for `groups` statement, eg a recursive translation in each `groups` with assuming that hierarchy as the root, but it's not supported yet.

Let me take a look anyways.

**Comment by codeout** [comment]
Hi @MicLennartz,

I've released v0.3.6 with `groups` statement support, I hope it also works for you.

```shell
$ echo "set groups VPN01894 snmp location 'room1'" | junoser -s
groups VPN01894 {
    snmp {
        location 'room1';
    }
}

$ junoser -v
0.3.6
```

**Comment by MicLennartz** [comment]
Hi @codeout ,
the new version looks very promising. Even with very complex config with and without groups it seem to be working well.

The only issue I found is, that 'set apply-groups ..." is not parsed and raises an exemption. But I guess, this is easy to fix.

**Comment by codeout** [comment]
Thanks, I'm done fixing it.
Could you double-check my code can parse your config before releasing the new version if you have time?

```shell
git clone https://github.com/codeout/junoser.git -b apply-groups
cd junoser

echo "set apply-groups foo" | ruby -I lib exe/junoser -s
# or
ruby -I lib exe/junoser -s your_config_file
```

**Comment by MicLennartz** [comment]
I've tested your new code and it seems like it can convert even very complex JunOS code now, in both directions.
Good work!

**Comment by codeout** [comment]
Just released v0.3.7 with the new code 🚀 Thank you for your testing!
